core/src/test/scala/org/apache/spark/scheduler/TaskSchedulerImplSuite.scala

```diff
@@ -27,18 +29,48 @@ class FakeSchedulerBackend extends SchedulerBackend {
   def defaultParallelism(): Int = 1
 }
 
-class TaskSchedulerImplSuite extends SparkFunSuite with LocalSparkContext with Logging {
+class TaskSchedulerImplSuite extends SparkFunSuite with LocalSparkContext with BeforeAndAfterEach
+    with Logging {
 
-  test("Scheduler does not always schedule tasks on the same workers") {
+
+  var failedTaskSetExc: Option[Throwable] = None
+  var failedTaskSetReason: String = null
+  var failedTaskSet = false
+
+  override def beforeEach(): Unit = {
+    super.beforeEach()
+    failedTaskSet = false
+    failedTaskSetExc = None
```

Comment:
super nit but could you write this out as "Exception"? I kept mentally reading it as "Executor" -- and it doesn't look like it will throw any lines over 100